### ISSUE TITLE
spaced-comment should control asterisk spacing and alignment in block comments

### ISSUE THREAD

**Issue by Yajo** [eslint/eslint#10793]
**What rule do you want to change?**
spaced-comment

**Does this change cause the rule to produce more or fewer warnings?**
more

**How will the change be implemented? (New option, new default behavior, etc.)?**
new options, and possibly new defaults

**Please provide some example code that this change will affect:**

```js
/*eslint spaced-comment: ["error", "always"]*/
var mything = {
    /**
     *Initialize the widget & parameters.
        *
    *@param {Object} parent contains the form view.
        * @param {String} name the name of the field.
        * @param {Object} record information about the database records.
        * @param {Object} options view options.
        */
    init: function (parent, name, record, options) {
        this._super(parent, name, record, options);
        this.init_params();
    },
}
```

**What does the rule currently do for this code?**
It says it's :+1: 

**What will the rule do after it's changed?**
As you can see, this rule is missing 2 basic features:

1. Checking asterisks are aligned.
2. Checking a space after the asterisk exists.

IMHO those should be true by default.

There's also the `/* eslint multiline-comment-style: ["error", "starred-block"] */` rule, but that doesn't control problem 2 and also what happens if I do want to force jsdocs to be OK, but I don't want to force converting consecutive single line comments into mutiline comments?

https://eslint.org/docs/developer-guide/code-conventions#multi-line-comments says this is required, but ESLint does not provide a way to enforce it AFAICS.

**Comment by platinumazure** [comment]
Hi @Yajo, thanks for the issue!

My initial thoughts:

* I'm :+1: to enhancing spaced-comment to look at spaces after leading asterisks on later lines of block comments.
* I'm :-1: to enhancing spaced-comment to check for alignment of leading asterisks on later lines of block comments. If multiline-comment-style is failing to do this, I would be :+1: to fixing that rule (and honestly that does feel like a bug).
* With regard to alignment of JSDoc comments specifically, I'm not sure if that needs a separate rule or if that could be covered in multiline-comment-style. If it's possible to cover in multiline-comment-style, I think that would be my preference.

Let's see what others on the team think.
